### DISCUSSION TITLE
What is the purpose of the io.smallrye:smallrye-open-api-spring artifact?

### DISCUSSION THREAD

**Discussion by codespearhead** [smallrye/smallrye-open-api#1870] [Q&A] [answered]
Is the [io.smallrye/smallrye-open-api-spring](https://mvnrepository.com/artifact/io.smallrye/smallrye-open-api-spring) artifact intended to be an OpenAPI specification generator for Spring Boot, similar to how [io.quarkus/quarkus-smallrye-openapi](https://mvnrepository.com/artifact/io.quarkus/quarkus-smallrye-openapi) works for Quarkus?

If so, how do you use it?

If given some pointers, I'm willing to work on a tutorial like [this one](https://quarkus.io/guides/openapi-swaggerui#expose-openapi-specifications) and work towards making it available on [Spring Initializr](https://start.spring.io/).

I've only managed to get the following output:

```yaml
# target/generated/openapi.yaml
---
openapi: 3.0.3
info:
  title: Generated API
  version: "1.0"
paths:
  /api/hello:
    get:
      summary: Hello endpoint
      description: Returns a greeting message
      responses:
        "200":
          description: OK
          content:
            '*/*':
              schema:
                type: string
```

To do that, I created a sample Spring Boot project:

```bash
mkdir sample
cd sample
curl https://start.spring.io/starter.zip \
    -d type=maven-project \
    -d dependencies=web \
    -o sample.zip
unzip sample.zip
```

Then added the following to the pom.xml:

```xml
<dependency>
	<groupId>io.smallrye</groupId>
	<artifactId>smallrye-open-api-spring</artifactId>
	<version>3.11.0</version>
</dependency>
<dependency>
	<groupId>io.smallrye</groupId>
	<artifactId>smallrye-open-api-maven-plugin</artifactId>
	<version>3.11.0</version>
</dependency>
```

And ran:

```bash
./mvnw io.smallrye:smallrye-open-api-maven-plugin:generate-schema
```

However, unlike its Quarkus equivalent, all the following endpoints returned `HTTP 404` upon running `./mvnw start-spring:run`:

1 . http://localhost:8080/openapi
2. http://localhost:8080/q/openapi
3. http://localhost:8080/swagger
4. http://localhost:8080/q/swagger

Getting it to work on [spring-projects/spring-petclinic](https://github.com/spring-projects/spring-petclinic) is the next step.

**Comment by phillip-kruger** [comment, marked as answer]
No, the io.smallrye/smallrye-open-api-spring artifactis intended to be an OpenAPI specification generator for Spring support in Quarkus.
